### PR TITLE
Merge settings icons and built-in icons

### PR DESCRIPTION
**Pull request by 2metres**
**User-Facing Changes**
None

**Description**
Remove duplication

### CODE REVIEW

File: packages/studio/src/index.ts [hunk 1]
```diff
@@ -32,6 +32,66 @@ export interface Time {
...
+  | "Shapes"
+  | "Share"
+  | "SouthEast"
+  | "Star"
+  | "Timeline"
+  | "Topic"
+  | "Walk"
+  | "World"
+  | never; // never has no effect here other than keeping the semicolon on a separate line for easier conflict resolution
+
```
Comment (2metres) [inline packages/studio/src/index.ts:94]: @foxymiles is there a way to avoid having to duplicate list?

File: packages/studio/src/index.ts [hunk 2]
```diff
@@ -421,46 +481,6 @@ export interface ExtensionModule {
   activate: ExtensionActivate;
 }
 
-export type SettingsIcon =
```
Comment (jtbandes) [inline packages/studio/src/index.ts:424]: Unfortunately this is now a public API that may be used by third-party extensions, so we can't change these names. If we want to change the icons, we'll need to map the old names to whatever new icons we choose.

**Review by jtbandes — CHANGES_REQUESTED** [review]

File: packages/studio/src/index.ts (hunk 2 above)
Comment (2metres) [reply packages/studio/src/index.ts:424]: Can we map the SettingsIcon to the RegisteredIconNames?

Comment (jtbandes) [reply packages/studio/src/index.ts:424]: Yep, see here

https://github.com/foxglove/studio/blob/main/packages/studio-base/src/components/SettingsTreeEditor/NodeEditor.tsx#L283

 https://github.com/foxglove/studio/blob/main/packages/studio-base/src/components/SettingsTreeEditor/icons.ts#L46

**Pull request closed without merging**